js/ethereum-utils/src/signature.definitions.ts

```diff
@@ -0,0 +1,158 @@
+import { EipDomainPayload } from './payloads';
+
+export const EIP712_DOMAIN_DEFINITION = {
+  EIP712Domain: [
+    {
+      name: 'name',
+      type: 'string',
+    },
+    {
+      name: 'version',
+      type: 'string',
+    },
+    {
+      name: 'chainId',
+      type: 'uint256',
+    },
+    {
+      name: 'verifyingContract',
+      type: 'address',
+    },
+  ],
+};
+
+// using pallet_revive test chain ID for now.
+export const EIP712_DOMAIN_DEFAULT: EipDomainPayload = {
+  name: 'Frequency',
+  version: '1',
+  chainId: '0x190f1b44',
+  verifyingContract: '0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC',
+};
+
+export const ADD_PROVIDER_DEFINITION = {
```

Comment:
Again, I don't think so, but I wonder if we need to version these in any way? I guess we can later add `_V2` on the end without issue.

Reply:
I would say, any structs that already are at "V2" should get `_V2` appended to the const name now.

Reply:
If it the only version based on our existing convention I didn't add any version but all versions `> 1` should have the version in the name unless there's been a mistake (just like what Joe suggested)